types.go

```diff
@@ -908,6 +935,37 @@ type (
 		NetAmount           PurchaseUnitAmount  `json:"net_amount"`
 		TotalRefundedAmount *PurchaseUnitAmount `json:"total_refunded_amount,omitempty"`
 	}
+
+	ReferralRequest struct {
+		TrackingID    string      `json:"tracking_id"`
+		Operations    []Operation `json:"operations,omitempty"`
+		Products      []string    `json:"products,omitempty"`
+		LegalConsents []Consent   `json:"legal_consents,omitempty"`
+	}
+
+	Operation struct {
+		Operation                string             `json:"operation"`
+		APIIntegrationPreference IntegrationDetails `json:"api_integration_preference,omitempty"`
+	}
+
+	IntegrationDetails struct {
+		RestAPIIntegration RestAPIIntegration `json:"rest_api_integration"`
+	}
+
+	RestAPIIntegration struct {
+		IntegrationMethod string            `json:"integration_method"`
+		IntegrationType   string            `json:"integration_type"`
+		ThirdPartyDetails ThirdPartyDetails `json:"third_party_details"`
```

Comment:
omitempty?

Reply:
Third Party Details are required if IntegrationType is third party (only options are `FIRST_PARTY` or `THIRD_PARTY`). Since this package is just about exclusively for third-party integration, I thought it made sense to make it required.